src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceEntityUpdater.java

```diff
@@ -100,34 +101,34 @@ public void onServiceIdUpdateRequest(@Observes ServiceIdUpdateRequest event) {
         return;
       }
       
-      if (event.isPriority()) {
-        queue.remove(event.getId());
-        queue.add(0, event.getId());
-      } else {
-        if (!queue.contains(event.getId())) {
-          queue.add(event.getId());
-        }
-      }
+      queue.add(event);
     }
   }
 
   @Timeout
   public void timeout(Timer timer) {
     if (!stopped) {
-      if (!queue.isEmpty()) {
-        updatePtvService(queue.remove(0));          
+      ServiceIdUpdateRequest updateRequest = queue.next();
+      if (updateRequest != null) {
+        updatePtvService(updateRequest);
       }
 
       startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
 
-  private void updatePtvService(ServiceId serviceId) {
+  private void updatePtvService(ServiceIdUpdateRequest updateRequest) {
+    updatePtvService(updateRequest.getId(), updateRequest.getOrderIndex());
+  }
+  
+  private void updatePtvService(ServiceId serviceId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updatePtvService" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)